lib/main.dart

```diff
@@ -1,6 +1,38 @@
 import 'package:flutter/material.dart';
+import 'dart:convert';
+import 'dart:async';
+import 'package:http/http.dart' as http;
+import 'package:flutter/foundation.dart';
+import 'catData.dart';
 
 void main() {
-  runApp(MyApp());
+  runApp(MainPage());
 }
 
+Future<List<CatData>> fetchCats(http.Client client) async {
+  final response = await client.get('https://api.thecatapi.com/v1/breeds/');
+  return compute(parseCat, response.body);
+}
+
+List<CatData> parseCat(responseBody) {
+  final parsed = jsonDecode(responseBody) as Map<String, dynamic>;
+  return parsed[''].map<CatData>((json) => CatData.fromJson(json)).toList();
+}
+
+class MainPage extends StatefulWidget {
```

Comment:
Maybe name AnimalsListScreen will be better?

Reply:
fixed